packages/core/src/utils/validate-sign-in-experience.ts

```diff
@@ -19,3 +25,26 @@ export const validateTermsOfUse = (termsOfUse: Optional<TermsOfUse>) => {
     'sign_in_experiences.empty_content_url_of_terms_of_use'
   );
 };
+
+const validateOneAndOnlyOnePrimarySignInMethod = (signInMethods: SignInMethods) => {
+  const signInMethodStates = Object.values(signInMethods);
+  const primarySignInMethodCount = signInMethodStates.reduce(
+    (primaryCount, signInMethodState) =>
+      primaryCount + (signInMethodState === SignInMethodState.primary ? 1 : 0),
+    0
+  );
+  assertThat(
+    primarySignInMethodCount === 1,
+    'sign_in_experiences.not_one_and_only_one_primary_sign_in_method'
+  );
+};
+
+export const validateSignInMethods = (signInMethods?: SignInMethods) => {
+  if (!signInMethods) {
+    return;
+  }
+
+  validateOneAndOnlyOnePrimarySignInMethod(signInMethods);
+
+  // TODO: assert others next PR
+};
```

Comment:
Could merge as one single`validateSignInMethods` ?

Reply:
there will five more validate function here and this function will be too long and not readable enough to understand quickly. we can decide whether to merge later next pr.

Reply:
Inlined. PTAL.